lib/cretonne/meta/isa/intel/encodings.py

```diff
@@ -260,8 +261,22 @@ def enc_i32_i64_ld_st(inst, w_bit, recipe, *args, **kwargs):
 # Function addresses.
 #
 
-I32.enc(base.func_addr.i32, *r.fnaddr4(0xb8))
-I64.enc(base.func_addr.i64, *r.fnaddr8.rex(0xb8, w=1))
+I32.enc(base.func_addr.i32, *r.fnaddr4(0xb8),
+        isap=Not(allones_funcaddrs))
+I64.enc(base.func_addr.i64, *r.fnaddr8.rex(0xb8, w=1),
+        isap=Not(allones_funcaddrs))
+
+I32.enc(base.func_addr.i32, *r.allones_fnaddr4(0xb8),
+        isap=allones_funcaddrs)
+I64.enc(base.func_addr.i64, *r.allones_fnaddr8.rex(0xb8, w=1),
+        isap=allones_funcaddrs)
```

Comment:
We generate slightly better encoding tables when you can put the predicates on the recipe instead of on the specific instruction. It doesn't make much of a difference.

Reply:
I looked into this, and there appear to be some bugs with predicates on recipes. I think I can fix it, but I'll do it in a separate patch because it's a little complex.

Reply:
I think @angusholder ran into the same problem in #160. ISTR he had a fix in that PR.